### PR TITLE
Add authentication and refresh token endpoint.

### PR DESCRIPTION
**Pull request by aaron-wang**
Two new endpoints added:
1. Exchange an authorization code for an **access token** and **refresh token**
2. Exchange a refresh token for an **access token**

After talking with Yu, we decided to not add any tests since we'd essentially be testing Google's token request functionality. 

**Note:** Set the client ID and client secret as environment variables before using these endpoints.

### CODE REVIEW

File: service/src/main/resources/api/openapi.yml [hunk 1]
```diff
@@ -35,12 +35,48 @@ paths:
               schema:
                 $ref: '#/components/schemas/VersionProperties'
 
+  /api/auth/exchangeAuthorizationCode:
+    parameters:
+      - $ref: '#/components/parameters/AuthCode'
+    post:
+      summary: |
+        Exchange an authorization code for an access token and refresh token.
+      operationId: exchangeAuthCode
```
Comment (yuhuyoyo) [inline service/src/main/resources/api/openapi.yml]: getAuthTokens

File: service/src/main/resources/api/openapi.yml [hunk 2]
```diff
@@ -35,12 +35,48 @@ paths:
               schema:
                 $ref: '#/components/schemas/VersionProperties'
 
+  /api/auth/exchangeAuthorizationCode:
+    parameters:
+      - $ref: '#/components/parameters/AuthCode'
+    post:
```
Comment (yuhuyoyo) [inline service/src/main/resources/api/openapi.yml]: get

File: service/src/main/resources/api/openapi.yml [hunk 3]
```diff
@@ -35,12 +35,48 @@ paths:
...
+          $ref: '#/components/responses/BadRequest'
+        '403':
+          $ref: '#/components/responses/PermissionDenied'
+        '500':
+          $ref: '#/components/responses/ServerError'
+
+  /api/auth/refreshToken:
+    parameters:
+      - $ref: '#/components/parameters/RefreshToken'
+    post:
```
Comment (yuhuyoyo) [inline service/src/main/resources/api/openapi.yml]: get

File: service/src/main/java/bio/terra/axonserver/app/controller/AuthorizationApiController.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+
+@Controller
+public class AuthorizationApiController implements AuthorizationApi {
+  private final HttpServletRequest servletRequest;
+
+  @Value("${AXON_UI_CLIENT_ID}")
```
Comment (yuhuyoyo) [inline service/src/main/java/bio/terra/axonserver/app/controller/AuthorizationApiController.java:23]: i think this is a fine pattern given this is the only place using these env vars. If we ever need to use them in multiple places, we should adapt the wsm pattern where we add these two fields to application.yaml profile and set them to these two environment variables.

**Review by yuhuyoyo — APPROVED** [review]

**Review by cbookg — APPROVED** [review]

**Pull request merged**